### PR TITLE
Timeline events improvements for displaying user and app details 

### PR DESCRIPTION
**Pull request by Cloud11PL**
## What type of PR is this?
- [ ] 💅 Refactor
- [x] 🌟 Feature
- [x] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Usage Instructions, Screenshots, Recordings

CleanShot 2024-04-12 at 10 27 33

CleanShot 2024-04-12 at 10 16 32

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

## Description

* Fixes an issue where avatar in history component shows incorrect initials
* Adds url to Avatar component if it exists on user
* Makes date format for timeline events consistent - the date will now have format of "X ... ago"
* Notes made by Apps will now show avatar (if available) or fallback to initials

### CODE REVIEW

File: src/components/Timeline/TimelineNote.tsx [hunk 1]
```diff
@@ -73,31 +70,50 @@ const NoteMessage: React.FC<NoteMessageProps> = ({ message }) => (
...
 );
 
+const TimelineAvatar = ({
+  user,
+  app,
+  classes,
+}: {
+  user: OrderEventFragment["user"];
+  app: OrderEventFragment["app"] | GiftCardEventFragment["app"];
+  classes: string;
```
Comment (poulch) [inline src/components/Timeline/TimelineNote.tsx:80]:
```suggestion
  className: string;
```

File: src/components/Timeline/TimelineNote.tsx [hunk 2]
```diff
@@ -73,31 +70,50 @@ const NoteMessage: React.FC<NoteMessageProps> = ({ message }) => (
...
+    );
+  }
+
+  return null;
+};
+
 export const TimelineNote: React.FC<TimelineNoteProps> = props => {
-  const { date, user, message, hasPlainDate } = props;
-  const { user: currentUser } = useUser();
+  const { date, user, message, hasPlainDate, app } = props;
```
Comment (poulch) [inline src/components/Timeline/TimelineNote.tsx:106]:
```suggestion
export const TimelineNote: React.FC<TimelineNoteProps> = ({ date, user, message, hasPlainDate, app }) => {
```

File: src/giftCards/GiftCardUpdate/GiftCardHistory/GiftCardHistory.tsx [hunk 3]
```diff
@@ -88,7 +88,8 @@ const GiftCardHistory: React.FC = () => {
                       user={user}
                       message={message}
                       key={id}
-                      hasPlainDate
+                      app={app}
+                      hasPlainDate={false}
```
Comment (poulch) [inline src/giftCards/GiftCardUpdate/GiftCardHistory/GiftCardHistory.tsx:92]: question: why did you change that, having there plain date was intended

File: src/components/Timeline/TimelineNote.tsx [hunk 4]
```diff
@@ -73,31 +70,50 @@ const NoteMessage: React.FC<NoteMessageProps> = ({ message }) => (
   </>
 );
 
+const TimelineAvatar = ({
```
Comment (poulch) [inline src/components/Timeline/TimelineNote.tsx:73]: question: don't you thingk about move this component to a separate file?

File: src/giftCards/GiftCardUpdate/GiftCardHistory/GiftCardHistory.tsx (hunk 3 above)
Comment (Cloud11PL) [reply src/giftCards/GiftCardUpdate/GiftCardHistory/GiftCardHistory.tsx:92]: How do I know if it's intended? I don't think giving user an exact date and time is better than "x days ago" (full date is available on hover).

File: src/components/Timeline/TimelineNote.tsx (hunk 4 above)
Comment (Cloud11PL) [reply src/components/Timeline/TimelineNote.tsx:73]: This file has multiple components. `TimelineAvatar` is not intended to be used anywhere else apart from `TimelineNote` so I think it can stay here "privately" 😄

**Review by poulch — DISMISSED** [review]

**Review by poulch — APPROVED** [review]

File: src/components/Timeline/TimelineNote.test.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          message="Note"
+          hasPlainDate={false}
+        />
+      </Wrapper>,
+    );
+
+    expect(screen.getByText("Test User")).toBeInTheDocument();
+    expect(screen.getByText("Note")).toBeInTheDocument();
+    expect(screen.getByText("TU")).toBeInTheDocument();
+    expect(screen.getByText("a few seconds ago")).toBeInTheDocument();
```
Comment (poulch) [inline src/components/Timeline/TimelineNote.test.tsx:33]: All test should follow AAA test conversion. I'll give you example please follow it in rest of tests
```suggestion
    // Arrange
    const mockedUser = { // Add type here
            avatar: null,
            id: "1",
            email: "test@test.com",
            firstName: "Test",
            lastName: "User",
            __typename: "User",
          }
          
     // Act
    render(
      <TimelineNote
        app={null}
        user={mockedUser}
        date={wrapperFriendlyDate}
        message="Note"
        hasPlainDate={false}
      />
      , { wrapper: Wrapper }
    );

    // Assert
    expect(screen.getByText("Test User")).toBeInTheDocument();
    expect(screen.getByText("Note")).toBeInTheDocument();
    expect(screen.getByText("TU")).toBeInTheDocument();
    expect(screen.getByText("a few seconds ago")).toBeInTheDocument();
```

**Review by poulch — APPROVED** [review]

File: src/components/Timeline/TimelineNote.test.tsx (hunk 5 above)
Comment (Cloud11PL) [reply src/components/Timeline/TimelineNote.test.tsx]: Thanks for the suggestion, I updated the tests

**Review by poulch — APPROVED** [review]

File: src/components/Timeline/TimelineNote.test.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        user={mockedUser}
+        date={wrapperFriendlyDate}
+        message="Note"
+        hasPlainDate={false}
+      />,
+      { wrapper: Wrapper },
+    );
+
+    // Assert
+    const avatar = container.querySelector("img");
```
Comment (poulch) [inline src/components/Timeline/TimelineNote.test.tsx:68]: suggestion: using `container` is not recommended, you could query this img by role or by tag name easily

**Review by poulch — APPROVED** [review]

**Review by Droniu — APPROVED** [review]
